src/ui/public/kuery/lib/ast.js

```diff
@@ -0,0 +1,58 @@
+import grammar from 'raw!./kuery.peg';
+import PEG from 'pegjs';
+import _ from 'lodash';
+
+const kueryParser = PEG.buildParser(grammar);
+
+export function fromKueryExpression(expression) {
+  if (_.isUndefined(expression)) {
+    throw new Error('expression must be a string, got undefined instead');
+  }
+
+  return kueryParser.parse(expression);
+}
+
+export function toKueryExpression(ast) {
+  if (!Array.isArray(ast)) throw new Error('AST must be an array');
+
+  return ast.reduce((acc, node) => {
+    return `${acc} ${stringifyNode(node)}`;
+  }, '').trim();
+}
+
+function stringifyNode(node) {
+  if (!node || !node.type) return '';
+
+  let retVal = '';
+  if (node.negate) retVal += '-';
+
+  switch (node.type) {
```

Comment:
You're referring to the code that converts the AST into an elasticsearch query? You're probably right, it does make sense to house all the conversion logic for a given language in the same place. Establishes a better pattern for third party languages